### DISCUSSION TITLE
Authentication failed for ftgres when trying to generate first alembic migration file on postgres

### DISCUSSION THREAD

**Discussion by jalvespinto** [sqlalchemy/sqlalchemy#7705] [Usage Questions]
I have a flask app using sqlalchemy with postgres. The app and db are both running on docker containers. Now I am trying to generate an initial migration file to create the tables, but I am getting the error `sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) FATAL:  password authentication failed for user "postgres" FATAL:  password authentication failed for user "postgres".` 
For alembic.ini I am using "sqlalchemy.url = postgresql:postgres:test@localhost:test_db". The postgres docker container is up and have port 54322 exposed. Using "docker exec db psql -U postgres" I can access postgres and see that 'test_db' is there and have postgres as owner. Also, I can connect to it using d-beaver. So I don't get why sqlachemy is not able to connect. I am new to alembic and for the target_metadata on alembic env file I am using `registry().metadata` since I am using imperative mapping. I guess the issue is not related to an error there, but since I don't get exactly how alembic works I decided it was worth mentioning. tks

**Comment by zzzeek** [comment]
you likely need to configure TCP /password authentication for your PostgreSQL database.  See https://www.postgresql.org/docs/current/auth-pg-hba-conf.html for how this is configured on PG server.

**Reply by CaselIT** [reply]
seems strange that dbeaver would work if that's the case, tough?

also please note that this connection string is not properly formatted: `postgresql:postgres:test@localhost:test_db` it's missing `//`

**Reply by jalvespinto** [reply]
Tks guy. Apparently it was just a missing port number (54322). I was following a tutorial that did not use the port number, but I guess is because they were using the standard 5432. I imagine in that case you don't need to type in the port.

**Reply by CaselIT** [reply]
Yes, the default port for postgresql if it's not specified is 5432
